### PR TITLE
Refactor proc macro channels

### PR DESCRIPTION
**Pull request by Draggu**
**Stack**:
- #9
- #8 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: src/lang/proc_macros/controller.rs [hunk 1]
```diff
@@ -256,19 +251,25 @@ fn parse_proc_macro_response(response: RpcResponse) -> Result<ProcMacroResult> {
 }
 
 #[derive(Clone)]
-pub struct ProcMacroChannelsReceivers {
+pub struct ProcMacroChannels {
```
Comment (piotmag769) [inline src/lang/proc_macros/controller.rs:254]: Imo we should have separate structs for senders and receivers (can be substructs too)

Comment (Draggu) [reply src/lang/proc_macros/controller.rs:254]: There is no need to do this, it would make it less clear only.

Comment (piotmag769) [reply src/lang/proc_macros/controller.rs:254]: I disagree, it is a weird struct with 2 pubs, but I can give it a pass

Comment (mkaput) [reply src/lang/proc_macros/controller.rs:254]: I'd agree with Piotr M, but indeed this looks like a style thing. Let's stay this as-is to not bike-shed.

**Review by mkaput — APPROVED** [review]

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**